client/components/parts/SignUpForm.jsx

```diff
@@ -0,0 +1,108 @@
+import React, {PureComponent} from 'react'
+import classNames from 'classnames'
+
+// use when state and lifecycle functions are needed
+export default class SignUpForm extends PureComponent {
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      siteInputState: 'empty',
+      /*
+      Site Input States:
+        - empty
+        - okay
+        - invalid
+        - unavailable
+        - testing
+      */
+      siteInput: '',
+      valid: false
+    }
+
+    this.timer
+  }
+
+
+  static propTypes = {
+    hostUrl: React.PropTypes.string
+  }
+
+  static defaultProps = {
+    hostUrl: 'uvic.club'
+  }
+
+  siteChange = (e) => {
+    const self = this
+    const site = e.target.value
+
+    this.setState({siteInput: site, siteInputState: 'testing'})
+
+    this.validateSite(site, function(s){ self.setState({siteInputState: s}) })
+  }
+
+  validateSite = (site, callback) => {
+    clearTimeout(this.timer) // remove
+    if (site === '') return callback('empty')
+    else if (!/^[a-zA-Z0-9-]+$/.test(site)) return callback('invalid')
+
+    // TODO: Test with the DB, This simulates a server call
+    this.timer = setTimeout(function() {
+      callback(site === 'taken' ? 'unavailable' : 'okay')
+    }, 200) // remove
+  }
+
+  siteInputHint = () => {
+    const siteInput = this.state.siteInput
+    switch (this.state.siteInputState) {
+      case 'okay':
+        return (<span><b className="siteName">{siteInput}.{this.props.hostUrl}</b> is available!</span>)
+      case 'unavailable':
+        return (<span><b className="siteName">{siteInput}.{this.props.hostUrl}</b> is unavailable.</span>)
+      case 'invalid':
+        return (<span><b className="siteName">{siteInput}.{this.props.hostUrl}</b> is invalid.</span>)
+      case 'testing':
+        return (<span>Checking <b className="siteName">{siteInput}.{this.props.hostUrl}</b>.</span>)
+      default:
+        return (<span>&nbsp;</span>)
```

Comment:
We should have 5 cases and a default which it falls into if there's an error (i.e. the siteInputState string doesn't match one of our expected strings)